### PR TITLE
Fix: crmd,tools: Set attributes for remote nodes directly into cib if it's legacy attrd

### PR DESCRIPTION
**Pull request by gao-yan**
Legacy attrd cannot handle attributes for remote nodes. Fix it by
conditionally reverting part of 488d82f.

### CODE REVIEW

File: crmd/utils.c [hunk 1]
```diff
@@ -1069,9 +1069,40 @@ update_attrd_helper(const char *host, const char *name, const char *value, const
...
+    if (is_remote_node) {
+        /* host is required for updating a remote node */
+        CRM_CHECK(host != NULL, return;);
+        /* remote node uname and uuid are equal */
+        if (update_without_attrd(host, name, value, user_name) < pcmk_ok) {
+            crm_err("Could not update attribute %s for remote-node %s", name, host);
+        }
+        return;
+    }
+#endif
```
Comment (kgaillot) [inline crmd/utils.c]: This belongs in update_attrd_helper().

Wouldn't you need to handle command='C' somehow as well?

I"m surprised someone wants to use pacemaker_remote with heartbeat. That's not exactly a well tested use case ...

File: tools/crm_resource_runtime.c [hunk 2]
```diff
@@ -652,8 +652,20 @@ cli_resource_delete(cib_t *cib_conn, crm_ipc_t * crmd_channel, const char *host_
...
+#else
+        if (node && is_remote_node(node)) {
+            /* Talk directly to cib for remote nodes if it's legacy attrd */
+            rc = delete_attr_delegate(cib_conn, cib_sync_call, XML_CIB_TAG_STATUS, node->details->id, NULL, NULL,
+                                  NULL, attr_name, NULL, FALSE, NULL);
+        } else {
+            rc = attrd_update_delegate(NULL, 'D', host_uname, attr_name, NULL, XML_CIB_TAG_STATUS, NULL,
+                                  NULL, NULL, node ? is_remote_node(node) : FALSE);
+        }
+#endif
```
Comment (kgaillot) [inline tools/crm_resource_runtime.c]: I think I'd prefer to functionize this so we can avoid duplicating the attrd_update_delegate() code (ie if !atomic ... delete_attr_delegate, return)

**Comment by gao-yan** [comment]
Amended. Thanks for the advice.

There are users that want to use pacemaker_remote with corosync-1 plugin-based cluster, since they are still running SLE11.  So far I haven't seen any other issues except with the legacy attrd. But indeed, we need more testing on it.

**Comment by kgaillot** [comment]
Looks great, thanks.

**Pull request merged**